src/utils/process.rs

```diff
@@ -1,6 +1,34 @@
 use std::collections::{HashMap, HashSet};
+use std::path::Path;
 use sysinfo::{Pid, Process, ProcessExt, SystemExt};
 
+use lazy_static::lazy_static;
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum CallingProcess {
+    GitGrep((HashSet<String>, HashSet<String>)),
+    OtherGrep, // rg, grep, ag, ack, etc
+}
+
+pub fn calling_process() -> Option<CallingProcess> {
+    #[cfg(not(test))]
+    {
+        CACHED_CALLING_PROCESS.clone()
+    }
+    #[cfg(test)]
+    {
+        determine_calling_process()
+    }
+}
```

Comment:
> I suggest abstracting access to that variable and not use `lazy_static` when testing.

@th1000s Ah of course (but I bet it would have taken me some time to realise), thanks!

Is this roughly what you had in mind? What would be the right approach for avoiding the `clone()` here? My first thought was to return `Option<Cow<'static, CallingProcess>>`, but I'm not sure how to deal with

> the trait bound `std::borrow::Cow<'_, _>: std::convert::From<&utils::process::CallingProcess>` is not satisfied

Reply:
(I'm thinking `Cow` is the right approach and have the `Cow` version compiling now)

Reply:
Nice, I would have reached for a macro there :)